utils/saasherder.py

```diff
@@ -245,35 +261,46 @@ def _collect_images(self, resource):
 
         return images
 
+    @staticmethod
+    def _check_image(image, image_patterns, image_auth, error_prefix):
+        error = False
+        if image_patterns and \
+                not any(image.startswith(p) for p in image_patterns):
+            error = True
+            logging.error(
+                f"{error_prefix} Image is not in imagePatterns: {image}")
+        try:
+            valid = Image(image, **image_auth)
+            if not valid:
+                error = True
+                logging.error(
+                    f"{error_prefix} Image does not exist: {image}")
+        except Exception as e:
+            error = True
+            logging.error(f"{error_prefix} Image is invalid: {image}. " +
+                          f"details: {str(e)}")
+
+        return error
+
     def _check_images(self, options):
         saas_file_name = options['saas_file_name']
         resource_template_name = options['resource_template_name']
         html_url = options['html_url']
-        resource = options['resource']
+        resources = options['resources']
         image_auth = options['image_auth']
         image_patterns = options['image_patterns']
         error_prefix = \
             f"[{saas_file_name}/{resource_template_name}] {html_url}:"
-        error = False
-        images = self._collect_images(resource)
 
-        for image in images:
-            if image_patterns and \
-                    not any(image.startswith(p) for p in image_patterns):
-                error = True
-                logging.error(
-                    f"{error_prefix} Image is not in imagePatterns: {image}")
-            try:
-                valid = Image(image, **image_auth)
-                if not valid:
-                    error = True
-                    logging.error(
-                        f"{error_prefix} Image does not exist: {image}")
-                    continue
-            except Exception:
-                error = True
-                logging.error(f"{error_prefix} Image is invalid: {image}")
-                continue
+        images_list = threaded.run(self._collect_images, resources,
+                                   self.available_thread_pool_size)
```

Comment:
did you test this without threading? just curious to see how much faster this is with the threaded...

Reply:
this actually doesn't add much :)

Reply:
I'd expect that :)